src/Features/CSharp/Portable/Completion/CompletionProviders/ObjectCreationCompletionProvider.cs

```diff
@@ -79,6 +79,17 @@ protected override (string displayText, string insertionText) GetDisplayAndInser
                 return (symbol.Name, symbol.Name);
             }
 
+            // If not all  of the type arguments are concrete types, eg
+            // we've inferred something like `Task<T>`, just suggest the
+            // generic type `Task<>`
+            var enclosingSymbol = context.SemanticModel.GetEnclosingSymbol(context.Position);
+            if (symbol is INamedTypeSymbol nt &&
+                nt.TypeArguments.Any(t => t is ITypeParameterSymbol typeParameter
+                    && !typeParameter.IsAccessibleFromSymbolOrEnclosingType(enclosingSymbol)))
```

Comment:
:memo: The capture of `enclosingSymbol` means this method gets at least two new allocations.

Reply:
This feels *super* weird that this is the place we're updating any logic...

Reply:
this is the wrong place for this logic.  The place we should be doing this is where we're actually figuring out the symbol to preselect.  once we figure out htat symbol, we should update things *there* to say if we actually need to use another symbol.  We do not need change the "display/insertion text logic".